Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -6,7 +6,9 @@ def fix_dynamic_linkage
 
         each_install_name_for(file) do |bad_name|
           # Don't fix absolute paths unless they are rooted in the build directory
-          next if bad_name.start_with?("/") && !bad_name.start_with?(HOMEBREW_TEMP.to_s)
+          next if bad_name.start_with?("/") &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.to_s) &&
+                  !bad_name.start_with?(HOMEBREW_TEMP.realpath.to_s)
```

Comment:
Yeh, forgot it'll only 💥 on bad symlinks in which case you'll have other problems.